libraries/chain/include/eosio/chain/hotstuff.hpp

```diff
@@ -54,7 +54,12 @@ namespace eosio::chain {
       quorum_certificate_message   high_qc; //justification
    };
 
-   using hs_message = std::variant<hs_vote_message, hs_proposal_message, hs_new_block_message, hs_new_view_message>;
+   using hs_message_inner = std::variant<hs_vote_message, hs_proposal_message, hs_new_block_message, hs_new_view_message>;
+
+   struct hs_message {
+      std::optional<uint32_t>     connection_id;
```

Comment:
I don't think this should be added to the message. This requires the `hs_message_inner` to be copied. Better to pass it along as extra params.

Reply:
Ok, will do that.